bee-hive/bee_hive/run_workflow.py

```diff
@@ -25,19 +31,129 @@ def load_agent(agent):
         agent_store = json.load(f)
     return agent_store[agent]
 
-
+# --- Abstract Base Class for Agent Runners ---
+class AgentRunner(ABC):
```

Comment:
We need a generic way to deal with exceptions. Right now it seems like you are just passing the exception to the caller. 

That’s one approach. Another approach to think about is an exception handler. So something like:

```python
class AbstractAgent(ABC):
  #…
  def exception_handler(self, handler) -> error:
     pass
  #…
```
We can then let the `handler` object be defined and passed by the caller. This will allow exceptions to be handled by caller of the runner. For instance if there was a policy to retry.

Also if the handler is not defined then by default the exception is propagated up as you are doing now.

Reply:
Agreed - inclined to promote this to an issue in it's own right so we can have a dedicated discussion on how we manage exceptions

Reply:
Will track in #134 - so we can get the basics refactored merged asap since the base code layout is changing.